keps/sig-node/4786-direct-cgroup-stats-collection/README.md

```diff
@@ -0,0 +1,527 @@
+    -   [ ](R) Minimum Two Week Window for GA e2e tests to prove flake free
+-   [ ](R) Graduation criteria is in place
+    -   [ ](R)
+        [all GA Endpoints](https://github.com/kubernetes/community/pull/1806)
+        must be hit by
+        [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+-   [ ](R) Production readiness review completed
+-   [ ](R) Production readiness review approved
+-   [ ] "Implementation History" section is up-to-date for milestone
+-   [ ] User-facing documentation has been created in [kubernetes/website], for
+    publication to [kubernetes.io]
+-   [ ] Supporting documentation—e.g., additional design documents, links to
+    mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The [cAdvisor] is still started and running behind the scene by the Kubelet
+after [KEP-2371] to gather system cgroup stats. Running cAdvisor brings in extra
+overhead and complexity. However, the stat collection of system cgroups is
+trivial to implement with a low overhead as there are only a handful of system
+cgroups.
+
+This KEP aims to eliminate the need to run cAdvisor with enablement of KEP-2371
+for better performance and simplicity.
+
+[cAdvisor]: https://github.com/google/cadvisor
+[KEP-2371]: https://github.com/kubernetes/enhancements/edit/master/keps/sig-node/2371-cri-pod-container-stats/README.md
+
+### Current Use of cAdvisor
+
+After enabling feature `PodAndContainerStatsFromCRI`, only
+[summary API][summary-api] invokes cAdvisor for stats of:
+
+*   Root filesystem
+*   Root cgroup (CPU, Memory, Swap and Network stats)
+*   System cgroup (CPU, Memory and Swap)
+    *   Kubelet cgroup
+    *   Runtime cgroup
+    *   Pod Root cgroup
+    *   Misc cgroup
+
+[summary-api]: https://github.com/kubernetes/kubernetes/blob/release-1.30/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go
+
+## Motivation
+
+Running cAdvisor is a visible performance overhead and there are work to tune
+the cAdvisor parameters to minimize the overhead such as [#18044] and [#124520].
+The overhead would be much higher on Nodes of high pod density as there are many
+more cgroups to collect stats, which is also redandunt when
+`PodAndContainerStatsFromCRI` is enabled since pod stats are collected via CRI
+APIs.
+
+[#18044]: https://github.com/kubernetes/kubernetes/issues/18044
+[#124520]: https://github.com/kubernetes/kubernetes/pull/124520
+
+### Goals
```

Comment:
is this work related: https://github.com/kubernetes/kubernetes/pull/107960?

Reply:
Seems related. Added.